### PR TITLE
SearchPaths improvements

### PR DESCRIPTION
**Pull request by est77**
- Remove default arguments from SearchPaths.
- Use the default platform separator when initializing SearchPaths from environment variables.
- Added reset method (needed by Gafferseed).
- Added method to split and push back paths separated by a separator (needed by Gafferseed).
- Added unit tests.

### CODE REVIEW

File: src/appleseed/foundation/meta/tests/test_searchpaths.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        const char* TestEnvironmentValue = "/tmp:/usr/tmp:/var/local/tmp";
+        const char TestSeparator = ':';
+
+#if defined _WIN32
+        _putenv_s(TestEnvironmentName, TestEnvironmentValue);
+#else
+        setenv(TestEnvironmentName, TestEnvironmentValue, 1);
+#endif
+        SearchPaths searchpaths(TestEnvironmentName, TestSeparator);
+        EXPECT_EQ(searchpaths.to_string(TestSeparator, false), TestEnvironmentValue);
```
Comment (dictoon) [inline src/appleseed/foundation/meta/tests/test_searchpaths.cpp]: Conventionally, the expected value should be the first argument to `EXPECT_EQ()`.

File: src/appleseed/foundation/utility/searchpaths.h [hunk 2]
```diff
@@ -125,9 +133,10 @@ class SearchPaths
 
     // Return a string with all the search paths separated by the specified separator,
     // optionally making them absolute and/or listing them in reverse order.
-    std::string to_string(
-        const char separator = ':',
-        const bool reversed = false) const;
+    std::string to_string(const char separator, const bool reversed) const;
+
+    // Return the default separator character for platform.
```
Comment (dictoon) [inline src/appleseed/foundation/utility/searchpaths.h]: *for the platform*

File: src/appleseed/renderer/modeling/shadergroup/shaderquery.cpp [hunk 3]
```diff
@@ -203,7 +203,7 @@ ShaderQuery::ShaderQuery(const char* search_path)
 ShaderQuery::ShaderQuery(const SearchPaths& search_paths)
   : impl(new Impl())
 {
-    impl->m_search_path = search_paths.to_string();
+    impl->m_search_path = search_paths.to_string(':', false);
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/shadergroup/shaderquery.cpp]: Why aren't we using the platform separator here?

File: src/appleseed/foundation/utility/searchpaths.h [hunk 4]
```diff
@@ -125,9 +133,10 @@ class SearchPaths
 
     // Return a string with all the search paths separated by the specified separator,
     // optionally making them absolute and/or listing them in reverse order.
-    std::string to_string(
-        const char separator = ':',
-        const bool reversed = false) const;
+    std::string to_string(const char separator, const bool reversed) const;
```
Comment (dictoon) [inline src/appleseed/foundation/utility/searchpaths.h]: Maybe having two methods would be clearer than having a boolean argument? E.g.

    to_string()
    to_string_reversed()

File: src/appleseed/foundation/meta/tests/test_searchpaths.cpp [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    TEST_CASE(SearchPathSplitAndPushBack)
+    {
+        const char* TestPaths = "/tmp:/usr/tmp:/var/local/tmp";
+        const char TestSeparator = ':';
+
+        SearchPaths searchpaths;
+        searchpaths.split_and_push_back(TestPaths, TestSeparator);
+        EXPECT_EQ(searchpaths.to_string(TestSeparator, false), TestPaths);
+    }
+}
```
Comment (dictoon) [inline src/appleseed/foundation/meta/tests/test_searchpaths.cpp]: I suggest adding tests for corner cases, at least for when `TestPaths` is an empty string.
